src/hotspot/share/gc/g1/g1CollectedHeap.hpp

```diff
@@ -168,6 +169,53 @@ class G1CollectedHeap : public CollectedHeap {
   // Testing classes.
   friend class G1CheckRegionAttrTableClosure;
 
+  class StalledAllocReq : public DoublyLinkedListNode {
```

Comment:
Please add a short comment about it; maybe it would even be useful to describe the lifecycle (state transitions) of the allocation request here.